xwiki-authenticator-trusted-adapters/xwiki-authenticator-trusted-attributes/src/main/java/org/xwiki/contrib/authentication/AttributesTrustedAuthenticationAdapter.java

```diff
@@ -0,0 +1,184 @@
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.xwiki.component.annotation.Component;
+
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.web.XWikiRequest;
+
+/**
+ * Implementation of the {@link TrustedAuthenticationAdapter} for attributes authentication.
+ *
+ * @since 1.9.0
+ */
+@Component
+@Singleton
+@Named("attributes")
+public class AttributesTrustedAuthenticationAdapter implements TrustedAuthenticationAdapter
+{
+    // Configuration
+    private static final String CONFIG_AUTH_FIELD = "auth_field";
+
+    private static final String CONFIG_ID_FIELD = "id_field";
+
+    private static final String CONFIG_GROUP_FIELD = "group_field";
+
+    private static final String CONFIG_GROUP_VALUE_SEPARATOR = "group_value_separator";
+
+    private static final String CONFIG_LOGOUT_URL = "logout_url";
+
+    // Default values for configuration
+    private static final String DEFAULT_AUTH_FIELD = "remote_user";
+
+    private static final String DEFAULT_ID_FIELD = DEFAULT_AUTH_FIELD;
+
+    private static final String DEFAULT_GROUP_VALUE_SEPARATOR = "\\|";
+
+    private static final String LOGOUT_URL_REDIRECTION_PLACEHOLDER = "__REDIRECT__";
+
+    private static final char COMMA_SEPARATOR = ',';
+
+    @Inject
+    private Logger logger;
+
+    @Inject
+    private TrustedAuthenticationConfiguration configuration;
+
+    @Inject
+    private Provider<XWikiContext> contextProvider;
+
+    @Override
+    public String getUserUid()
+    {
+        // There is no need for a Secret field when attributes.
+        // Secret field cross checking is a pseudo security fix for the relative less secure HTTP headers.
+        return getAttribute(configuration.getCustomProperty(CONFIG_AUTH_FIELD, DEFAULT_AUTH_FIELD));
+    }
+
+    @Override
+    public String getUserName()
+    {
+        return getAttribute(configuration.getCustomProperty(CONFIG_ID_FIELD, DEFAULT_ID_FIELD));
+    }
+
+    @Override
+    public String getUserProperty(String name)
+    {
+        return getAttribute(name);
+    }
+
+    @Override
+    public boolean isUserInRole(String role)
+    {
+        return getUserRoles().contains(role);
+    }
+
+    /**
+     * @param name the name of the attribute.
+     * @return the value of the named request attribute, or null if no value is defined.
+     */
+    private String getAttribute(String name)
+    {
+        if (StringUtils.isBlank(name)) {
+            return null;
+        }
+
+        XWikiRequest request = contextProvider.get().getRequest();
+        if (request == null) {
+            return null;
+        }
+
+        try {
+            value = (String) request.getAttribute(name); // returns an Object
+            return value.trim(); // Just to cleanup any whitespace which should not be needed
+        } catch (ClassCastException e) {
+            logger.debug("Failed to cast attribute [{}] as string.", name, e);
+            return null;
+        } catch (NullPointerException e) {
```

Comment:
Not a fan of this kind of (very slow) way to handle problems like these. Would be much cleaner to actively check if the value is a `String` before manipulating it.

Reply:
I updated it with a simpler flow!